### PR TITLE
[P2P / Integration Tests] Remove Task.Delay and optimise burst mode

### PR DESCRIPTION
**Pull request by fassadlr**
Please see comments in #3183

### CODE REVIEW

File: src/Stratis.Bitcoin/P2P/PeerConnector.cs [hunk 1]
```diff
@@ -272,6 +266,12 @@ internal async Task ConnectAsync(PeerAddress peerAddress)
             }
         }
 
+        [NoTrace]
+        private TimeSpan CalculateConnectionInterval()
+        {
+            return this.ConnectorPeers.Count < this.ConnectionSettings.BurstModeTargetConnections ? TimeSpan.Zero : TimeSpans.Second;
```
[the remark this answers is not in the document]
Comment (fassadlr) [reply src/Stratis.Bitcoin/P2P/PeerConnector.cs]: OK yes, I will make this abstract and move it into the individual connectors.

[the remark this answers is not in the document]
Comment (zeptin) [reply src/Stratis.Bitcoin/P2P/PeerConnector.cs]: See https://github.com/stratisproject/StratisBitcoinFullNode/pull/1946
Also see issue https://github.com/stratisproject/StratisBitcoinFullNode/issues/1131

File: src/Stratis.Bitcoin/P2P/PeerConnector.cs [hunk 2]
```diff
@@ -272,6 +266,15 @@ internal async Task ConnectAsync(PeerAddress peerAddress)
             }
         }
 
+        /// <summary>
+        /// Determines how often the connector should try and connect to an address from it's list.
+        /// </summary>
+        [NoTrace]
+        public virtual TimeSpan CalculateConnectionInterval()
+        {
+            return this.ConnectorPeers.Count < this.ConnectionSettings.BurstModeTargetConnections ? TimeSpans.Ms100 : TimeSpans.Second;
```
Comment (dangershony) [inline src/Stratis.Bitcoin/P2P/PeerConnector.cs]: Should we not also rename this param if burst mode is gone now?

Comment (fassadlr) [reply src/Stratis.Bitcoin/P2P/PeerConnector.cs]: OK Yeah I have renamed this InitialConnectionTarget 👍

**Comment by dangershony** [comment]
Tested locally works fine

**Pull request merged**